source/Camera/Advanced/CaptureHDRPrintNormals/CaptureHDRPrintNormals.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+Capture Zivid point clouds, calculate normals and print a subset.
+
+For scenes with high dynamic range we combine multiple acquisitions to get an HDR point cloud.
+*/
+
+#include <Zivid/Zivid.h>
+
+#include <iostream>
+
+namespace
+{
+    void printNormals(int radius,
+                      const Zivid::Array2D<Zivid::NormalXYZ> &normals,
+                      const int num_of_rows,
+                      const int num_of_cols)
+    {
+        const auto lineSep = std::string(50, '-');
+        std::cout << lineSep << std::endl;
+        for(int row = (num_of_rows / 2 - radius); row < (num_of_rows / 2 + radius); row++)
+        {
+            for(int col = (num_of_cols / 2 - radius); col < (num_of_cols / 2 + radius); col++)
+            {
+                std::cout << "Normals[" << row << "," << col << "]: ";
+                std::cout << normals(row, col) << std::endl;
```

Comment:
Can you post how the printout looks like?

I haven't tested it but it could be nice if it is similar to https://github.com/zivid/zivid-cpp-samples/blob/cc2828d041e407340c82e30fe2b2cc43d9fc0fa4/source/Applications/Basic/FileFormats/ReadIterateZDF/ReadIterateZDF.cpp

Reply:
Connecting to camera
Configuring settings
Adding acquisition with aperture = 9.57
Adding acquisition with aperture = 4.76
Adding acquisition with aperture = 2.59
Capturing frame (HDR)
Computing normals and copying them to CPU
Printing normals for the central 10 x 10
--------------------------------------------------
Normals[595,955]: { x: nan, y: nan, z: nan }
Normals[595,956]: { x: 0.455703, y: -0.890131, z: 0.00121597 }
Normals[595,957]: { x: nan, y: nan, z: nan }
Normals[595,958]: { x: nan, y: nan, z: nan }
Normals[595,959]: { x: -0.486586, y: -0.373692, z: -0.789676 }
Normals[595,960]: { x: 0.176712, y: -0.978236, z: -0.108755 }
Normals[595,961]: { x: -0.57049, y: -0.621963, z: -0.53638 }
Normals[595,962]: { x: -0.310799, y: -0.830651, z: -0.461978 }
Normals[595,963]: { x: nan, y: nan, z: nan }
Normals[595,964]: { x: 0.222549, y: -0.178329, z: -0.958473 }
Normals[596,955]: { x: nan, y: nan, z: nan }
Normals[596,956]: { x: nan, y: nan, z: nan }
Normals[596,957]: { x: nan, y: nan, z: nan }
Normals[596,958]: { x: nan, y: nan, z: nan }
Normals[596,959]: { x: 0.0687043, y: -0.947002, z: -0.313795 }
Normals[596,960]: { x: nan, y: nan, z: nan }
Normals[596,961]: { x: -0.398845, y: -0.627832, z: -0.668393 }
Normals[596,962]: { x: nan, y: nan, z: nan }
Normals[596,963]: { x: -0.637359, y: -0.412933, z: -0.650584 }
Normals[596,964]: { x: nan, y: nan, z: nan }
Normals[597,955]: { x: nan, y: nan, z: nan }
Normals[597,956]: { x: nan, y: nan, z: nan }
Normals[597,957]: { x: nan, y: nan, z: nan }
Normals[597,958]: { x: nan, y: nan, z: nan }
Normals[597,959]: { x: 0.163095, y: -0.930057, z: -0.329231 }
Normals[597,960]: { x: 0.226796, y: -0.914879, z: -0.334006 }
Normals[597,961]: { x: -0.236145, y: -0.805623, z: -0.543329 }
Normals[597,962]: { x: -0.342579, y: -0.709652, z: -0.615657 }
Normals[597,963]: { x: -0.356435, y: -0.659211, z: -0.662114 }
Normals[597,964]: { x: -0.556702, y: -0.30985, z: -0.770763 }

Reply:
We also limit the number of decimal places to 1